### PR TITLE
adapt spl, rms & energy to work with Signal and FramedSignal

### LINKED ISSUE #114: Sound Pressure Level and RMS should be computable on FramedSignals
Right now, it is only possible to compute SPL and RMS on the whole signal. It should also work on `FramedSignal` instances, where the function returns a SPL/RMS value for each frame.

### PR DESCRIPTION
**Pull request by superbock**

### CODE REVIEW

**Comment by superbock** [comment]
@fdlm, does this fix your original issue (#114)?

File: madmom/audio/signal.py [hunk 1]
```diff
@@ -323,10 +322,9 @@ def trim(signal, where='fb'):
...
 
-def root_mean_square(signal):
+def energy(signal):
     """
-    Computes the root mean square of the signal. This can be used as a
-    measurement of power.
+    Compute the energy of the signal.
 
     Parameters
     ----------
```
Comment (fdlm) [inline madmom/audio/signal.py]: Adapt Parameters docstring (signal can also be a `FramedSignal`)

File: madmom/audio/signal.py [hunk 2]
```diff
@@ -335,23 +333,51 @@ def root_mean_square(signal):
...
+
+
+def root_mean_square(signal):
+    """
+    Compute the root mean square of the signal. This can be used as a
+    measurement of power.
+
+    Parameters
+    ----------
+    signal : numpy array
```
Comment (fdlm) [inline madmom/audio/signal.py]: Adapt Parameters docstring (signal can also be a `FramedSignal`)

File: madmom/audio/signal.py [hunk 3]
```diff
@@ -335,23 +333,51 @@ def root_mean_square(signal):
...
+    return np.sqrt(energy(signal) / signal.size)
 
 
 def sound_pressure_level(signal, p_ref=None):
     """
-    Computes the sound pressure level of a signal.
+    Compute the sound pressure level of a signal.
 
     Parameters
     ----------
```
Comment (fdlm) [inline madmom/audio/signal.py]: Adapt Parameters docstring (signal can also be a `FramedSignal`)

**Pull request merged**